client/client_test.go

```diff
@@ -78,3 +84,78 @@ func equals(tb testing.TB, exp, act interface{}) {
 		tb.FailNow()
 	}
 }
+
+func getExtensions() map[string]string {
+	// ensure we at least return the extensions the first time we get called
```

Comment:
@unleashed just a fyi, its  seen as good practice to mark these functions as test helpers https://golang.org/pkg/testing/#T.Helper

Reply:
ah, just merged - I am not happy about having these in the code package, and I wonder whether we can remove then from it so that this code is only compiled in when tests are run. Does this mechanism do that?

Reply:
I see it's just a "helper" for cleaner debugging output. I think we'll want to check whether having the tests and helpers here is good practice.

Reply:
No worries I was just pointing it out, didn't spot it in original review. 

AFAIK, by default, files with `_test.go` are not compiled until the test is run. Also when you build a binary or import some package as a dependency, the `_test.go` files will not be included.

Reply:
> I think we'll want to check whether having the tests and helpers here is good practice.

I have seen some talks and blogs though that say they prefer to include all code per test, even it means duplication because it makes the test easier to follow top to bottom. I think the approach here is fine though tbh.